ibis/backends/flink/registry.py

```diff
@@ -40,9 +40,21 @@ def extract_field_formatter(translator: ExprTranslator, op: ops.Node) -> str:
 
 
 def _cast(translator: ExprTranslator, op: ops.generic.Cast) -> str:
-    if op.to.is_timestamp() and op.to.timezone:
-        arg_translated = translator.translate(op.arg)
-        return f"TO_TIMESTAMP(CONVERT_TZ(CAST({arg_translated} AS STRING), 'UTC+0', '{op.to.timezone}'))"
+    from ibis.expr.datatypes.core import JSON, Timestamp
+
+    arg, to = op.arg, op.to
+    arg_translated = translator.translate(arg)
+    if isinstance(to, Timestamp):
+        if arg.dtype.is_numeric():
+            arg_translated = f"FROM_UNIXTIME({arg_translated})"
+
+        if to.timezone:
+            return f"TO_TIMESTAMP(CONVERT_TZ(CAST({arg_translated} AS STRING), 'UTC+0', '{to.timezone}'))"
+        else:
+            return f"TO_TIMESTAMP({arg_translated})"
+
+    elif isinstance(to, JSON):
```

Comment:
Don't check with `isinstance`, use the is_* methods. A method exists for every leaf type as well as every parent class.

```suggestion
    from ibis.expr.datatypes.core import JSON, Timestamp

    arg, to = op.arg, op.to
    arg_translated = translator.translate(arg)
    if to.is_timestamp():
        if arg.dtype.is_numeric():
            arg_translated = f"FROM_UNIXTIME({arg_translated})"

        if to.timezone:
            return f"TO_TIMESTAMP(CONVERT_TZ(CAST({arg_translated} AS STRING), 'UTC+0', '{to.timezone}'))"
        else:
            return f"TO_TIMESTAMP({arg_translated})"

    elif to.is_json():
```

Reply:
Thanks, I replaced `isinstance()` with `is_*` in all my changes.